src/vivarium_public_health/metrics/mortality.py

```diff
@@ -7,121 +7,173 @@
 excess mortality in the simulation.
 
 """
+from collections import Counter
+from typing import Dict, List
+
+import pandas as pd
+
+from vivarium.framework.engine import Builder, ConfigTree
+from vivarium.framework.event import Event
+from vivarium.framework.population import PopulationView
+from vivarium.framework.time import Timedelta
+
 from vivarium_public_health.disease import DiseaseState, RiskAttributableDisease
-from vivarium_public_health.metrics.utilities import (
-    get_age_bins,
-    get_deaths,
-    get_person_time,
-    get_years_of_life_lost,
-)
+from vivarium_public_health.metrics.stratification import ResultsStratifier
+from vivarium_public_health.utilities import to_time_delta
 
 
 class MortalityObserver:
     """An observer for cause-specific deaths, ylls, and total person time.
 
-    By default, this counts cause-specific deaths, years of life lost, and
-    total person time over the full course of the simulation. It can be
-    configured to bin these measures into age groups, sexes, and years
-    by setting the ``by_age``, ``by_sex``, and ``by_year`` flags, respectively.
+    By default, this counts cause-specific deaths and years of life lost over
+    the full course of the simulation. It can be configured to add or remove
+    stratification groups to the default groups defined by a ResultsStratifier.
```

Comment:
Use `:class:ResultsStratifier` so the docs link to the the stratifier class when built.